openedx_lti_tool_plugin/templates/openedx_lti_tool_plugin/course_outline.html

```diff
@@ -0,0 +1,119 @@
+{% load i18n %}
+
+<div id='course_title' class='h3'></div>
+<hr>
+<div id='outline'></div>
+
+<style>
+  #outline .fa-folder-o {
+    color: var(--bs-gray-500);
+    padding-right: 0.5rem;
+  }
+  .sequential-item {
+    padding-left: 2rem !important;
+  }
+  .vertical-item {
+    padding-left: 3rem !important;
+  }
+</style>
+
+{{ course_outline|json_script:'course_outline' }}
+<script>
+  /**
+   * Create list HTML element.
+   *
+   * @param {Object} block - Block object.
+   * @param {string} [collapseId = null] - Collapse element ID.
+   *
+   * @return {HTMLElement} - List item HTML element.
+   */
+  function createListItem(block, collapseId = null) {
+    // Create icon element.
+    const icon = Object.assign(document.createElement('i'), {
+      className: 'fa fa-folder-o',
+      ariaHidden: 'true',
+    });
+    // Create list item element.
+    const listItem = Object.assign(document.createElement('a'), {
+      innerHTML: `${icon.outerHTML} ${block.display_name}`,
+      className: `${block.type}-item list-group-item list-group-item-action`,
+    });
+
+    // Set required collapse button attributes if collapseId is present.
+    if (collapseId !== null) {
+      listItem.setAttribute('href', `#${collapseId}`);
+      listItem.setAttribute('data-bs-toggle', 'collapse');
+      listItem.setAttribute('aria-controls', collapseId);
+      listItem.setAttribute('aria-expanded', 'false');
+    };
+
+    return listItem;
+  };
+
+  /**
+   * Create collapsible HTML element.
+   *
+   * @param {Object} block - Block object.
+   * @param {string} collapseId - Collapse element ID.
+   *
+   * @return {HTMLElement} - Collapsible HTML element.
+   */
+  function createCollapse(block, collapseId) {
+    const collapseElem = Object.assign(document.createElement('div'), {
+      id: collapseId,
+      className: 'list-group collapse',
+    });
+
+    // Set chapter list collapsed by default.
+    if (block.type == 'chapter') {
+      collapseElem.classList.add('show');
+    };
+
+    return collapseElem;
+  };
+
+  // Get base URl and app name.
+  const baseUrl = '{{ request.scheme }}://{{ request.get_host }}';
+  const appName = '{{ request.resolver_match.app_name }}';
+
+  // Get course outline object, outline element and update element class.
+  const outline = JSON.parse(document.getElementById('course_outline').textContent);
+  const outlineElem = document.getElementById('outline');
+  outlineElem.classList.add('list-group');
+
+  // Get course title element and update text to course title.
+  const courseTitleElem = document.getElementById('course_title');
+  courseTitleElem.textContent = outline.display_name;
+
+  // Check that course contains chapters.
+  if (outline.children !== null) {
+    // Iteate over each course outline chapter.
+    outline.children.forEach(function callback(chapter, chapterIndex) {
```

Comment:
Why did you decide for the chapters to not contain the sequences?

Reply:
@kuipumu I'm not sure what do you mean, the chapters contain the sequences, what these lines do is check that the course contains any chapter and then iterate over each of its sequences. There could be a scenario where the course is launched but contains no sequences

Reply:
![image](https://github.com/Pearson-Advance/openedx-lti-tool-plugin/assets/62396424/925a1ff1-7794-4b09-badf-a3f1fb25507a)

When it's rendered it appears it's not containing the sequences

Reply:
@alexjmpb that's because the button that enabled or disables the collapse list cannot be inside it, the same applies to each child element: https://getbootstrap.com/docs/5.0/components/collapse/